lib/command.js

```diff
@@ -1094,6 +1094,24 @@ Expecting one of '${allowedValues.join("', '")}'`);
     return this;
   }
 
+  /**
+   * Note: masked in some unit tests.
+   * @param {string} executableFile
+   * @param {string} executableDir
+   * @param {string} subcommandName
+   */
+  _throwForMissingExecutable(executableFile, executableDir, subcommandName) {
```

Comment:
The method name is `_throwForMissingExecutable`, but it may not throw, so why not name it something like `_validateExecutable`?

Or how about doing the checks outside of the method?
```js
if (!fs.existsSync(executableFile)) {
 _throwForMissingExecutable(...);
}
```

Reply:
> method name

Oh, yes. I originally had the check outside the routine. Will change name.

> Or how about doing the checks outside of the method?

I agree that would be tidier for the code and usage. However, I started that way, and multiple unit tests check whether `fs.existsSync` has been called and broke. I didn't think of an easy way of fixing the tests, other than moving the `fs. fs.existsSync` inside the routine. Then easy to completely bypass the new code and run the old tests.

Reply:
We have several `_checkForX` routines already so following that style, `_checkForMissingExecutable`.

Reply:
Lots!
- `_checkForMissingMandatoryOptions`
- `_checkForConflictingLocalOptions`
- `_checkForConflictingOptions`
- `_checkForBrokenPassThrough`